pandas/tests/io/parser/common/test_ints.py

```diff
@@ -110,6 +116,85 @@ def test_integer_overflow_bug(all_parsers, sep):
     tm.assert_frame_equal(result, expected)
 
 
+def _iinfo(dtype):
+    pdtype = pandas_dtype(dtype)
+    iinfo = np.iinfo(pdtype.type if is_extension_array_dtype(dtype) else pdtype)
+    return iinfo
+
+
+_raises_any_integer_cast_exception = pytest.raises(  # noqa: PDF010
+    (OverflowError, TypeError, ValueError),
+    match="|".join(
+        [
+            "Overflow",
+            "cannot safely cast non-equivalent",
+            "Integer out of range",
+            "Unable to convert column",
+            "The elements provided in the data cannot all be casted to the dtype",
+        ]
+    ),
+)
+
+
+@skip_pyarrow
+@pytest.mark.parametrize(
+    "getval,expected",
+    [
+        (lambda dtype: _iinfo(dtype).max, nullcontext()),  # in range does not raise
+        (lambda dtype: _iinfo(dtype).min, nullcontext()),  # in range does not raise
+        (lambda dtype: _iinfo(dtype).max + 1, _raises_any_integer_cast_exception),
+        (lambda dtype: _iinfo(dtype).min - 1, _raises_any_integer_cast_exception),
+    ],
+)
+def test_integer_overflow_with_user_dtype(all_parsers, any_int_dtype, getval, expected):
+    dtype = any_int_dtype
+    parser = all_parsers
+    val = getval(dtype)
+    data = f"A\n{val}"
+
+    # Positive value overflow with uint8, uint16, uint32 and any overflow with
```

Comment:
You can remove that comment I think.

Also please add gh reference to your tests

Reply:
Okay, see [92fab59](https://github.com/pandas-dev/pandas/pull/47168/commits/92fab5955375267548c80848fdd622f53731c474)